build-reporter-plugin.js

```diff
@@ -12,7 +12,13 @@ class BugsnagBuildReporterPlugin {
     const plugin = (compilation, cb) => {
       const stats = compilation.getStats()
       if (stats.hasErrors()) return cb(null)
-      reportBuild(this.build, this.options)
+      const options = Object.assign(this.options,
+        !this.options.logger && compiler.getInfrastructureLogger
+          ? { logger: compiler.getInfrastructureLogger('BugsnagBuildReporterPlugin') }
```

Comment:
consider removing duplication of the plugin name string (it is repeated further down)